modules/travel_pay/app/controllers/travel_pay/application_controller.rb

```diff
@@ -38,21 +38,8 @@ def after_logger
       logger.info('travel-pay') { Utils::Logger.build(self).after }
     end
 
-    def authorize
-      auth_header = request.headers['Authorization']
-      raise_unauthorized('Missing Authorization header') if auth_header.nil?
-      raise_unauthorized('Authorization header missing Bearer token') unless auth_header.start_with?('Bearer ')
-    end
-
-    def raise_unauthorized(detail)
-      raise Common::Exceptions::Unauthorized.new(detail:)
-    end
-
-    # Blocks requests from being handled if feature flag is disabled
-    def block_if_flag_disabled
-      unless Flipper.enabled?(:travel_pay_power_switch, @current_user)
-        raise Common::Exceptions::ServiceUnavailable, detail: 'This feature has been temporarily disabled'
-      end
+    def feature_enabled
+      routing_error unless Flipper.enabled?(:travel_pay_power_switch, @current_user)
```

Comment:
I disagree with this change to the raised error, but I'm willing to talk about it. I can see it argued both ways, but my biggest concern is that I don't know what status code comes back with the `routing_error`. 

The RoutingError class doesn't specify one, which makes me think it defaults to a 500 or something. The ServiceUnavailable is a known status code that is for temporarily unavailable services, which gives us some intrinsic info about why we're getting the error.

Let's chat though, happy to hear your thoughts.

Reply:
I don't have strong feelings either way, so I'm fine reverting this to the service unavailable error

Reply:
After reflecting on this more, I don't think a `5xx` code is really appropriate. [503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) means that a service is unavailable, but really, we want to say "this is unavailable _to you_, the current requester". I think the most appropriate code in this case would actually be a [403](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403).